### PR TITLE
User Improved Search Results

### PR DESCRIPTION
**Pull request by Danwhy**
### To be merged after #346

Adds basic logging of how many times users click on search results.

Stores number of clicks in leveldb and provides prefixes to keys to easily identify 'count groups', so results with counts between certain values (say 10000 and 100000) are grouped together for easy extraction from leveldb, and then given a score boost.

### CODE REVIEW

File: lib/scores.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
+const Async = require('async');
```
Comment (SimonLab) [inline lib/scores.js]: A wiki page for this file will be great

**Comment by SimonLab** [comment]
Just some doc to add to explain how we define the boost depending on the count, otherwise LGTM

**Comment by Danwhy** [comment]
Added some documentation here https://github.com/TheScienceMuseum/collectionsonline/wiki/User-Improved-Search-Results

File: lib/scores.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const level = require('../lib/level.js');
```
Comment (nelsonic) [inline lib/scores.js]: @Danwhy bearing in mind that this ties the _score_ to a single server I would _recommend_ you consider asking @jamieu if we can use Redis (ElastiCache): http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/Clusters.Create.Redis.CON.html to ensure _horizontal scalability_ ...

Comment (Danwhy) [inline lib/scores.js]: @nelsonic  We definitely were thinking about using Redis eventually, Alan just suggested LevelDB as a simple way to get started without having to worry about deploying anything new. We've got an open issue reminding us to consider other options when scaling: https://github.com/TheScienceMuseum/collectionsonline/issues/339, I'll add the Redis suggestion

**Comment by nelsonic** [comment]
@Danwhy the code looks (_really_) good. 
_However_ I would _recommend_ that you use Redis instead of LevelDB to ensure that on-instance data is not lost in a crash and we can add nodes to the ELB cluster...
I _know_ that _initially_ we won't "_need_" to have multiple Node.js nodes ... but it would _good_ to have "_failover_" ...
Thoughts? Was this discussed on an issue? (_sorry to give you more work..._)

File: lib/scores.js (hunk 2 above)
Comment (nelsonic) [inline lib/scores.js]: Cool. ok. as long as the team are _aware_ that we either need to implement "_sync_" between the node.js servers (_if they use LevelDB_) or use ElastiCache to store the data, I'm happy with either. 👍

**Comment by Danwhy** [comment]
I'd be very happy to use Redis for this. Will just have to ask Jamie about getting it up and running

**Comment by nelsonic** [comment]
@Danwhy let me know if you need anything from me to make progress with this. 👍

**Comment by Danwhy** [comment]
@nelsonic So we've decided to leave this for now, as @jamieu is going to look into using the google analytics for the tracking, and then we can use that to boost the relevance. The code may be useful in the future though, so I'd like to keep the branch.  
I'll add a link to the branch in the relevant issue and close this pr.

**Comment by jamieu** [comment]
I agree, leave for now and close issue, but keep the branch.

I can already see how we might use / build upon this in future for
scenarios where google analytics doesn't cut it.

**Pull request closed without merging**